docs/consume-packages/consuming-packages-authenticated-feeds.md

```diff
@@ -3,98 +3,106 @@ title: Consuming packages from authenticated feeds
 description: Consuming packages from authenticated feeds in all NuGet client scenarios
 author: nkolev92
 ms.author: nikolev
-ms.date: 02/28/2020
+ms.date: 12/22/2023
 ms.topic: conceptual
 ---
 
 # Consuming packages from authenticated feeds
 
-In addition to the nuget.org [public feed](https://api.nuget.org/v3/index.json), NuGet clients have the ability to interact with file feeds and private http feeds.
+Many NuGet operations, such as restore and install, require communication with one or more package sources, which [can be configured in *nuget.config* files](../reference/nuget-config-file.md#packagesources).
+For HTTP feeds, NuGet will with make an unauthenticated request, and if the server responds with an HTTP 401 response, NuGet will search for credentials in the following order:
 
+1. [An environment variable `NuGetPackageSourceCredentials_{name}`](#credentials-in-environment-variables).
+1. [Credentials in *nuget.config* files](#credentials-in-nugetconfig-files).
+1. [Use a NuGet credential provider, if your package source provides one](#credential-providers).
 
-To authenticate with private http feeds, the 2 approaches are:
+> [!NOTE]
+> We recommend using a credential provider when possible.
+> Secrets will not be saved in the *nuget.config* file, reducing risk of accidentally leaking secrets.
+> Additionally, it typically reduces the number of places you need to update when a credential expires or changes.
+> If the credential provider supports single sign on, it may reduce the number of time you need to log in, or the number of places that credentials need to be saved.
 
-* Add credentials in the [NuGet.config](../reference/nuget-config-file.md#packagesourcecredentials)
-* Authenticate using one of the many extensibility models depending on the client used.
+The credentials you need to use are determined by the package source.
+Therefore, unless you're using a credential provider, you should check with your package source what credentials to use.
+It is very common for package sources to forbid using your password (that you log into the website with) with NuGet.
+Typically you need to create a Personal Access Token to use as NuGet's password, but you should check the documentation for the NuGet server you're using.
+Some package sources, such as Azure DevOps and GitHub, have scoped access tokens, so you may need to ensure that any tokens you create include the required scope.
 
-## NuGet clients' authentication extensibility
+## Credentials in environment variables

```

Comment:
I've always found environment variables as configurations to really reduce transparency since they are usually the highest priority in any config mode, and it's really difficult to know where something is configured.

I think that should ideally be represented, but I'd love to hear opinions.